app/assets/javascripts/commons.coffee

```diff
@@ -88,145 +86,116 @@ jQuery(document).ready ($) ->
-
-    # Set defaults when adding something dynamic to the form with cocoon
-    form.on 'cocoon:after-insert', (e, item) ->
-      if item.hasClass 'js-payment'
-        # default amount is what's unpaid
-        amount_item = item.find 'input[name*=amount]'
-        get_amounts controller_name, form, (data) ->
-          amount = data.gross_amount - data.paid_amount
-          amount = if amount > 0 then amount else 0
-          amount_item.val Math.round(amount*10**data.precision)/10**data.precision
-          return
-        # default date is today
-        date_item = item.find 'input[name*=date]'
-        date_item.val (new Date).toISOString().substr 0, 10
-      else if item.hasClass 'js-item'
-        init_invoice_item_autocomplete item.find('[data-role="item-description"]')
-        item.find('[data-role="taxes-selector"]').trigger('update')
-        autosize item.find('textarea')
-
-    # UX for taxes-selector labels
-    form.on 'click', '[data-role="taxes-selector-label"]', (e) ->
-      $(this).closest('.js-item').find('.select2-search__field').click()
-
-    # Execute actions when something dynamic is removed from the form
-    # with cocoon
-    form.on 'cocoon:before-remove', (e, item) ->
-      if item.hasClass 'js-item'
-        destroy_invoice_item_autocomplete item
-
-    # Set the autocomplete for customer selection
-    model = form.data('model')
-    $("##{model}_name").autocomplete {
-      source: '/customers/autocomplete.json',
-      select: (event, ui) ->
-        # Once the customer is selected autofill fields:
-        $("##{model}_customer_id").val ui.item.id
-        $("##{model}_identification").val ui.item.identification
-        $("##{model}_email").val ui.item.email
-        $("##{model}_contact_person").val ui.item.contact_person
-        $("##{model}_invoicing_address").val ui.item.invoicing_address
-        $("##{model}_shipping_address").val ui.item.shipping_address
-        autosize.update $('textarea.autosize')
-    }
-
-    # Configure Tax Selector behavior
-    $('[data-role="taxes-selector"]').select2()
-    $('#js-items-table').on 'cocoon:after-insert', (e, insertedItem) ->
-      $(insertedItem).find('[data-role="taxes-selector"]').select2()
-
-    form
-      .on 'update', '[data-role="taxes-selector"]', () ->
-        checked_taxes = $(this).find(':checked')
-        label = []
-        checked_taxes.each () ->
-          label.push $(this).closest('.checkbox').text().trim()
-        total = if label.length > 1 then ('(' + label.length + ')') else ''
-        label = if label.length > 0 then label.join(', ') else 'None'
-        $(this).find('[data-role="label"]').html(label)
-        $(this).find('[data-role="total"]').html(total)
-      .on 'change', '[data-role="taxes-selector"] :checkbox', (e) ->
-        $(this).closest('[data-role="taxes-selector"]').trigger('update')
-      .find('[data-role="taxes-selector"]').trigger('update')
+
+  form = $('form[data-role="invoice"]')
```

Comment:
actually all that code is inside a jQuery(document).ready ($) -> so variables are not global. Why should I wrap it with another function? What are those weird errors?